pkg/webhook/admission/servingruntime/servingruntime_webhook.go

```diff
@@ -0,0 +1,182 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package servingruntime
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/kserve/kserve/pkg/constants"
+	"net/http"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"strings"
+
+	"github.com/kserve/kserve/pkg/apis/serving/v1alpha1"
+
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+var log = logf.Log.WithName(constants.ServingRuntimeValidatorWebhookName)
+
+const (
+	InvalidPriorityError                      = "Same priority assigned for the model format %s"
+	InvalidPriorityServingRuntimeError        = "%s in the servingruntimes %s and %s in namespace %s"
+	InvalidPriorityClusterServingRuntimeError = "%s in the clusterservingruntimes %s and %s"
+)
+
+// +kubebuilder:webhook:verbs=create;update,path=/validate-serving-kserve-io-v1alpha1-clusterservingruntime,mutating=false,failurePolicy=fail,groups=serving.kserve.io,resources=clusterservingruntimes,versions=v1alpha1,name=clusterservingruntime.kserve-webhook-server.validator
+
+type ClusterServingRuntimeValidator struct {
+	Client  client.Client
+	Decoder *admission.Decoder
+}
+
+// +kubebuilder:webhook:verbs=create;update,path=/validate-serving-kserve-io-v1alpha1-servingruntime,mutating=false,failurePolicy=fail,groups=serving.kserve.io,resources=servingruntimes,versions=v1alpha1,name=servingruntime.kserve-webhook-server.validator
+
+type ServingRuntimeValidator struct {
+	Client  client.Client
+	Decoder *admission.Decoder
+}
+
+func (sr *ServingRuntimeValidator) Handle(ctx context.Context, req admission.Request) admission.Response {
+	servingRuntime := &v1alpha1.ServingRuntime{}
+	if err := sr.Decoder.Decode(req, servingRuntime); err != nil {
+		log.Error(err, "Failed to decode serving runtime", "name", servingRuntime.Name, "namespace", servingRuntime.Namespace)
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	ExistingRuntimes := &v1alpha1.ServingRuntimeList{}
+	if err := sr.Client.List(context.TODO(), ExistingRuntimes, client.InNamespace(servingRuntime.Namespace)); err != nil {
+		log.Error(err, "Failed to get serving runtime list", "namespace", servingRuntime.Namespace)
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+
+	// Only validate for priority if the new serving runtime is not disabled
+	if servingRuntime.Spec.IsDisabled() {
+		return admission.Allowed("")
+	}
+
+	for _, runtime := range ExistingRuntimes.Items {
+		if err := validateServingRuntimePriority(&servingRuntime.Spec, &runtime.Spec, servingRuntime.Name, runtime.Name); err != nil {
```

Comment:
@sivanantha321 Generally we do not take the address of the loop variable as it is reused in each iteration, see
https://www.evanjones.ca/go-gotcha-loop-variables.html